### ISSUE TITLE
Adding support for VM based runtimes

### ISSUE THREAD

**Issue by mcastelino** [intel/vhost-user-net-plugin#2]
Today when invoking the CNI plugin the return value typically looks like

```
{
  "ip4": {
    "ip": "10.56.217.133/24",
    "gateway": "10.56.217.1",
    "routes": [
      {
        "dst": "0.0.0.0/0"
      }
    ]
  },
  "dns": {}
}
```

The vhost socket information is passed in via volumes in 

```
/var/lib/cni/vhostuser/
```

However for VM based runtimes which can consume the vhost-user socket directly, sending out an additional value say
```
   "vhostfd": "/path/to/vhost/fd"
```
would be helpful.

A runtime would be free to ignore fields it did not recognize. 
golang json unmarshalling skips fields it does not recognize.

**Comment by rkamudhan** [comment]
@mcastelino Are you using clear containers or hyperd in your kubernetes cluster? I worked in both container runtime long back, like to get your idea on it.

**Comment by mcastelino** [comment]
@rkamudhan yes on Clear Containers. We already have support for vhost-user with our own CNM plugin for VPP. https://github.com/clearcontainers/vpp.

However here we setup a dummy interface to send in the information.

With CNI it should be much cleaner, specially since unrecognized fields are skipped.
Ideally extending CNI results field would be better. There is a thread about it based on another requirement

https://github.com/containernetworking/cni/issues/513#issuecomment-341147424

**Comment by rkamudhan** [comment]
oh cool, got your idea on it. Is that runtime not designed to ignore additional fields? I may be wrong, runtime care for IPAM information only right?

**Comment by mcastelino** [comment]
I meant json unmarshalling will drop that field so that runtimes that do not know or care about this field can ignore them till the spec gets standardized. They will not even see the field

https://play.golang.org/p/JTQTlWG5Ub
